src/main/java/hudson/tasks/junit/JUnitResultArchiver.java

```diff
@@ -142,11 +150,19 @@ protected TestResult parse(String expandedTestResults, AbstractBuild build, Laun
     @Override
     public void perform(Run build, FilePath workspace, Launcher launcher,
             TaskListener listener) throws InterruptedException, IOException {
+        TestResultAction action = parseAndAttach(this, null, build, workspace, launcher, listener);
+
+        if (action != null && action.getResult().getFailCount() > 0)
+            build.setResult(Result.UNSTABLE);
```

Comment:
I waffled on this a lot. This is something that I'd like to differently, but until https://issues.jenkins-ci.org/browse/JENKINS-43995 lands, we don't have a way to pick up that `UNSTABLE` status to cover the whole build later (modulo `try/catch` and the like). So for the moment, I'd say go with principle of least surprise: if you use `junit` now, it sets the build to `UNSTABLE`, so `junitResults` should do the same.